tests/v2/validation/upgrade/kubernetes.go

```diff
@@ -1,151 +1,188 @@
 	"time"
 
 	"github.com/sirupsen/logrus"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
-	apiv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
 	v1 "github.com/rancher/shepherd/clients/rancher/v1"
 	"github.com/rancher/shepherd/extensions/clusters"
 	"github.com/rancher/shepherd/extensions/clusters/bundledclusters"
+	"github.com/rancher/shepherd/extensions/defaults"
 	kcluster "github.com/rancher/shepherd/extensions/kubeapi/cluster"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/upgradeinput"
 	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
-	local = "local"
-
-	// isCheckingCurrentCluster is the boolean to decide whether validations should happen with a new GET request and cluster data or not
-	isCheckingCurrentCluster = true
-	// logMessageKubernetesVersion is the string message to log with the kubernetes versions validations' assertions
-	logMessageKubernetesVersion = "Validating the current version is the upgraded one"
-	// logMessageNodepoolVersion is the string message to log the with kubernetes nodepools versions validations' assertions
-	logMessageNodepoolVersions = "Validating the current nodepools version is the upgraded one"
+	local     = "local"
+	namespace = "fleet-default"
+	provider  = "provider.cattle.io"
+	rke       = "rke"
+	rke2      = "rke2"
 )
 
-func waitUntilLocalStable(client *rancher.Client, clusterName string) error {
-	return kwait.Poll(10*time.Second, 75*time.Minute, func() (bool, error) {
-		isConnected, err := client.IsConnected()
-		if err != nil {
-			logrus.Errorf("[connection stable]: %v", err)
-			return false, nil
-		}
-		logrus.Infof("[connection stable]: %v", isConnected)
+// upgradeLocalCluster is a function to upgrade a local cluster.
+func upgradeLocalCluster(u *suite.Suite, testName string, client *rancher.Client, clusterName string, testConfig *clusters.ClusterConfig, cluster upgradeinput.Cluster) {
+	clusterObject, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(u.T(), err)
+
+	clusterResp, err := client.Management.Cluster.ByID(clusterObject)
+	require.NoError(u.T(), err)
+
+	if clusterResp.Labels[provider] == rke {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.RKE1KubernetesVersions[0]
+		testName += " RKE1 cluster to " + testConfig.KubernetesVersion
+	} else if clusterResp.Labels[provider] == rke2 {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.RKE2KubernetesVersions[0]
+		testName += "Local cluster to " + testConfig.KubernetesVersion
+	} else {
+		testConfig.KubernetesVersion = cluster.ProvisioningInput.K3SKubernetesVersions[0]
+		testName += "Local cluster to " + testConfig.KubernetesVersion
+	}
```

Comment:
For rke, the test name mentiones `RKE1`.

For rke2 + k3s, the test name mentions `Local`.

Question(s):
- Should rke1 test name mention `local` as well?
- or Should rke2/k3s mention `rke2/k3s` instead of `local`?
- or -- is this as intended? 🤔

Reply:
Good catch, that is meant to have `local` across the board. I will update that.

Reply:
Resolved